src/misc/setup.cpp

```diff
@@ -838,18 +851,26 @@ Section_prop::~Section_prop()
 		delete (*prop);
 }
 
-Section_line* Config::AddSection_line(char const * const _name,void (*_initfunction)(Section*)) {
+Section_line *Config::AddSection_line(char const *const _name, SectionFunction func)
+{
 	Section_line* blah = new Section_line(_name);
-	blah->AddInitFunction(_initfunction);
+	blah->AddInitFunction(func);
 	sectionlist.push_back(blah);
 	return blah;
 }
 
+void Config::Init() const
+{
+	for (const auto &sec : sectionlist)
+		sec->ExecuteEarlyInit();
 
-void Config::Init() {
-	for (const_it tel=sectionlist.begin(); tel!=sectionlist.end(); ++tel) {
-		(*tel)->ExecuteInit();
-	}
+	for (const auto &sec : sectionlist)
+		sec->ExecuteInit();
```

Comment:
This code is OK, but I suggest writing `const auto *sec` (to avoid a pointer being passed by reference (therefore having two pointers)).

Reply:
Either you fixed this (and I cherry-picked it) or it gets mopped up in a later :-)